src/tests/BuildWasmApps/Wasm.Build.Tests/WasmRunOutOfAppBundleTests.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+#nullable enable
+
+namespace Wasm.Build.Tests
+{
+    public class WasmRunOutOfAppBundleTests : BuildTestBase
+    {
+        public WasmRunOutOfAppBundleTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext) : base(output, buildContext)
+        {}
+
+        [Theory]
+        [BuildAndRun(aot: false, host: RunHost.V8)]
+        [BuildAndRun(aot: true, host: RunHost.V8)]
+        [BuildAndRun(aot: false, host: RunHost.Chrome)]
+        [BuildAndRun(aot: true, host: RunHost.Chrome)]
+        [BuildAndRun(aot: false, host: RunHost.NodeJS)]
+        [BuildAndRun(aot: true, host: RunHost.NodeJS)]
+        public void RunOutOfAppBundle(BuildArgs buildArgs, RunHost host, string id)
+        {
+            buildArgs = buildArgs with { ProjectName = $"outofappbundle_{buildArgs.Config}_{buildArgs.AOT}" };
+            buildArgs = ExpandBuildArgs(buildArgs);
+
+            BuildProject(buildArgs,
+                            id: id,
+                            new BuildProjectOptions(
+                                InitProject: () => File.WriteAllText(Path.Combine(_projectDir!, "Program.cs"), s_mainReturns42),
+                                DotnetWasmFromRuntimePack: !(buildArgs.AOT || buildArgs.Config == "Release"),
+                                UseCache: false));
```

Comment:
Try removing this. Basically, what we do is build once for a config+aot-setting, and then use the same build to run with different hosts. So, it needs the cache for that.

Reply:
When using cache, the `AppBundle` directory was generated in a single nested directory.

Reply:
Would the test still be valid if you were to copy the `AppBundle` to a new temporary directory, and try to run from there? that would simulate running from a different directory too, right?

And what does `--deep-work-dir` do? It seems to be a nop currently.

Reply:
You’re right.
It looks like the test can be done by copying it from the AppBundle directory to another temporary directory and running it.

The `--deep-work-dir` is used to change the entry point in BuildTestBase for the execution target.
https://github.com/dotnet/runtime/pull/69441#discussion_r930099873

Reply:
Instead of `--deep-work-dir` stuff, add a new param - `jsRelativePath` defaulting to `test-main.js`, to `RunAndTestWasmApp`.